Deleting unexisting queue causes NPE

Whereas the method should be **idempotent**.
Stacktrace:
```
Caused by: java.lang.NullPointerException: null
	at com.github.fridujo.rabbitmq.mock.MockNode.queueDelete(MockNode.java:90)
	at com.github.fridujo.rabbitmq.mock.MockChannel.queueDelete(MockChannel.java:323)
	at com.github.fridujo.rabbitmq.mock.MockChannel.queueDelete(MockChannel.java:318)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.amqp.rabbit.connection.CachingConnectionFactory$CachedChannelInvocationHandler.invoke(CachingConnectionFactory.java:1140)
	at com.sun.proxy.$Proxy86.queueDelete(Unknown Source)
	at org.springframework.amqp.rabbit.core.RabbitAdmin.lambda$deleteQueue$3(RabbitAdmin.java:315)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.invokeAction(RabbitTemplate.java:2135)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.doExecute(RabbitTemplate.java:2094)
```

--- a/src/main/java/com/github/fridujo/rabbitmq/mock/MockNode.java
+++ b/src/main/java/com/github/fridujo/rabbitmq/mock/MockNode.java
@@ -86,9 +86,9 @@ public class MockNode implements ReceiverRegistry, TransactionalOperations {
     }
 
     public AMQP.Queue.DeleteOk queueDelete(String queueName, boolean ifUnused, boolean ifEmpty) {
-        MockQueue queue = queues.remove(queueName);
-        queue.notifyDeleted();
-        return new AMQImpl.Queue.DeleteOk(queue != null ? queue.messageCount() : 0);
+        Optional<MockQueue> queue = Optional.ofNullable(queues.remove(queueName));
+        queue.ifPresent(MockQueue::notifyDeleted);
+        return new AMQImpl.Queue.DeleteOk(queue.map(MockQueue::messageCount).orElse(0));
     }
 
     public AMQP.Queue.BindOk queueBind(String queueName, String exchangeName, String routingKey, Map<String, Object> arguments) {

--- a/src/test/java/com/github/fridujo/rabbitmq/mock/ChannelTest.java
+++ b/src/test/java/com/github/fridujo/rabbitmq/mock/ChannelTest.java
@@ -7,6 +7,7 @@ import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Random;
+import java.util.UUID;
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.TimeUnit;
@@ -169,6 +170,17 @@ class ChannelTest {
     }
 
     @Test
+    void exchangeDelete_does_nothing_when_not_existing() throws IOException, TimeoutException {
+        try (Connection conn = new MockConnectionFactory().newConnection()) {
+            try (Channel channel = conn.createChannel()) {
+                assertThat(channel.exchangeDelete(UUID.randomUUID().toString())).isNotNull();
+                assertThat(channel.exchangeDelete(UUID.randomUUID().toString(), false)).isNotNull();
+                channel.exchangeDeleteNoWait(UUID.randomUUID().toString(), false);
+            }
+        }
+    }
+
+    @Test
     void exchangeBind_binds_two_exchanges() throws IOException, TimeoutException {
         try (Connection conn = new MockConnectionFactory().newConnection()) {
             try (Channel channel = conn.createChannel()) {
@@ -288,6 +300,17 @@ class ChannelTest {
                 channel.queueDeleteNoWait("", false, false);
                 assertThatExceptionOfType(IOException.class)
                     .isThrownBy(() -> channel.queueDeclarePassive(""));
+            }
+        }
+    }
+
+    @Test
+    void queueDelete_does_nothing_when_not_existing() throws IOException, TimeoutException {
+        try (Connection conn = new MockConnectionFactory().newConnection()) {
+            try (Channel channel = conn.createChannel()) {
+                assertThat(channel.queueDelete(UUID.randomUUID().toString())).isNotNull();
+                assertThat(channel.queueDelete(UUID.randomUUID().toString(), false, false)).isNotNull();
+                channel.queueDeleteNoWait(UUID.randomUUID().toString(), false, false);
             }
         }
     }
@@ -601,7 +624,7 @@ class ChannelTest {
                 channel.txCommit();
                 assertThat(channel.basicGet(queue, true)).isNotNull();
                 // Channel contained only one message as transactions are cleared after commit
-                assertThat(channel.basicGet(queue, true)).isNull(); 
+                assertThat(channel.basicGet(queue, true)).isNull();
 
                 channel.basicPublish("", queue, null, "third message".getBytes());
                 assertThat(channel.basicGet(queue, true)).isNull();

